### PR TITLE
Update BUILD.md with shared provider information

### PR DESCRIPTION
**Pull request by RyanUnderhill**
**Description**: We should document how some providers are shared and behave differently. This adds information on how it works from a build/usage perspective.

### CODE REVIEW

**Review by jywu-msft — DISMISSED** [review]

File: BUILD.md [hunk 1]
```diff
@@ -177,6 +178,24 @@ Nuget packages are created under <native_build_dir>\nuget-artifacts
...
+On Windows, shared provider libraries will be named 'onnxruntime_providers_\*.dll' (for example onnxruntime_providers_openvino.dll).
+On Unix, they will be named 'libonnxruntime_providers_\*.so'
+On Mac, they will be named 'libonnxruntime_providers_\*.dylib'.
+
+There is also a shared library that shared providers depend on called onnxruntime_providers_shared (with the same naming convension applied as above).
+
+Note: It is not recommended to put these libraries in a system location or added to a library search path (like LD_LIBRARY_PATH on Unix). If multiple versions of onnxruntime are installed on the system this can make them find the wrong libraries and lead to undefined behavior.
+
+### Loading the shared providers
+Shared provider libraries are loaded by the onnxruntime code (do not load or depend on them in your client code). The API for loading shared or non shared providers is identical, the difference is that shared ones will be loaded at runtime when the provider is added to the session options (through a call like OrtSessionOptionsAppendExecutionProvider_OpenVINO or SessionOptionsAppendExecutionProvider_OpenVINO in the C API).
```
Comment (pranavsharma) [inline BUILD.md:194]:
```suggestion
Shared provider libraries are loaded by the onnxruntime code (do not load or depend on them in your client code). The API for registering shared or non shared providers is identical, the difference is that shared ones will be loaded at runtime when the provider is added to the session options (through a call like OrtSessionOptionsAppendExecutionProvider_OpenVINO or SessionOptionsAppendExecutionProvider_OpenVINO in the C API).
```

File: BUILD.md [hunk 2]
```diff
@@ -177,6 +178,24 @@ Nuget packages are created under <native_build_dir>\nuget-artifacts
...
+If a shared provider library cannot be loaded (if the file doesn't exist, or it's dependencies don't exist) then an error will be returned.
```
Comment (pranavsharma) [inline BUILD.md:195]:
```suggestion
If a shared provider library cannot be loaded (if the file doesn't exist, or its dependencies don't exist or not in the path) then an error will be returned.
```

**Comment by pranavsharma** [comment]
LGTM 👍 . 
I'm not why it's showing me as co-author. Hence I'll refrain from signing-off.

**Comment by RyanUnderhill** [comment]
> LGTM 👍 .
> I'm not sure why it's showing me as co-author. Hence I'll refrain from signing-off.

I think it's because I accepted your suggestions directly. It's just being smart.

**Review by jywu-msft — APPROVED** [review]

**Pull request merged**